### ISSUE TITLE
stm32f103xb_stm32f429zi_if uvision project builds, but does not get to main()

### ISSUE THREAD

**Issue by jhgorse** [ARMmbed/DAPLink#632]
Hello,

I am trying to work on porting the stm32f103xb_stm32f429zi_if project to GCC for the NUCLEO-F429ZI. I have followed the directions in the developer guide,
https://github.com/ARMmbed/DAPLink/blob/master/docs/DEVELOPERS-GUIDE.md
The NUCLEO board is modified such that the SWD pins are available for the interface STM32F103CBT chip (DAPLink) AND the STM32F429ZI is still connected to the SWD out and uart lines of the interface chip. See Figure 1 below for wiring diagram:
![image](https://user-images.githubusercontent.com/1783474/57962582-abb3a880-78e6-11e9-8575-e06cd9962fc8.png)

The project builds without complaint in Keil. I've removed a good deal of the code and it does not get to main(), so I suspect I have a configuration issue. Figure 2 below is the hang spot specifically:
![image](https://user-images.githubusercontent.com/1783474/57962547-0a2c5700-78e6-11e9-8488-f5905fe46496.png)

I have tried ST-Link V2 and J-Link Ultra+ probes via SWD.

The latest builds crash Windows 7 with the non-booting code by plugging it in via the USB to the interface.

The stm32f103xb_bl project builds without complaint, flashes, and runs as expected. It enumerates as HID CMSIS-DAP, CDC, and MSC composite device. The relevant difference is not obvious.

What can I do to get this working for the NUCLEO-F429ZI?

**Comment by jhgorse** [comment]
Referencing https://github.com/ARMmbed/DAPLink/pull/257

**Comment by jhgorse** [comment]
TL;DR: **don't use the crc .bin builds** for the stm32f103xb.

While directly programming the Main Firmware for a specific target should be possible, it is really meant to be loaded via the *_bl bootloader interface: taking the crc *.bin file and dropping it onto the MAINTENANCE USB MSC drive.

I tried the stm32f103xb_stm32f429zi_if_crc.bin, stm32f103xb_stm32f429zi_if_crc_legacy_0x5000.bin, and stm32f103xb_stm32f429zi_if_crc_legacy_0x8000.bin output files. The legacy 0x8000 was out of bounds. The others either succeeded in flashing and went into a unusable state (status LED at toggling at full speed) OR failed and reset back to MAINTENANCE.

I then tried stm32f103xb_stm32f429zi_if.bin. This works.

Closing the issue.
